internal/cloud/state.go

```diff
@@ -248,6 +253,8 @@ func (s *State) ShouldPersistIntermediateState(info *local.IntermediateStatePers
 		return true
 	}
 
+	// This value is controlled by a x-terraform-snapshot-interval header intercepted during
+	// state-versions API responses
 	if !s.enableIntermediateSnapshots && info.RequestedPersistInterval == time.Duration(0) {
```

Comment:
You should remove `&& info.RequestedPersistInterval == time.Duration(0)`, otherwise applies will always fail this condition